lib/rml-generator.js

```diff
@@ -166,21 +170,158 @@ class RMLGenerator extends AbstractGenerator {
+          namedNode(namespaces.rdf + 'type'),
+          namedNode(namespaces.td + 'PropertyAffordance')
+        ));
+
+        this.quads.push(quad(
+          propertyAffordanceSubject,
+          namedNode(namespaces.td + 'hasForm'),
+          formSubject
+        ));
+
+        this.quads.push(quad(
+          formSubject,
+          namedNode(namespaces.rdf + 'type'),
+          namedNode(namespaces.td + 'Form')
+        ));
+
+        this.quads.push(quad(
+          formSubject,
+          namedNode(namespaces.hctl + 'hasTarget'),
+          literal(source.access)
+        ));
+
+        this.quads.push(quad(
+          formSubject,
+          namedNode(namespaces.hctl + 'forContentType'),
+          literal(source.contentType)
+        ));
+
+        let operationType = null;
+        switch(source.operationType) {
+          case 'read':
+            operationType = namedNode(namespaces.td + 'readproperty');
+            break
+            
+          case 'write':
+          default:
+            break
+        }
+
+        this.quads.push(quad(
+          formSubject,
+          namedNode(namespaces.hctl + 'hasOperationType'),
+          operationType
+        ));
+
+        this.quads.push(quad(
+          wotSubject,
+          namedNode(namespaces.td + 'hasPropertyAffordance'),
+          propertyAffordanceSubject
+        ));
 
-      this._generateDatabaseDescription(databaseSubject, source);
+        // FIXME: W3C WoT Protocol Bindings
+
+        /* Build wotsec:$SCHEMESecurityScheme */
+        if (source.security) {
+          if (Array.isArray(source.security)) {
+            Object.keys(source.security).forEach(index => {
+              let security = source.security[index]
+              // FIXME: support more security schemes
+              if (security.type == 'apikey') {
+                this.quads.push(quad(
+                  securitySubject,
+                  namedNode(namespaces.rdf + 'type'),
+                  namedNode(namespaces.td + 'APISecurityScheme')
+                ));
+
+                this.quads.push(quad(
+                  securitySubject,
+                  namedNode(namespaces.wotsec + 'in'),
+                  literal(security.in)
+                ));
+
+                this.quads.push(quad(
+                  securitySubject,
+                  namedNode(namespaces.wotsec + 'name'),
+                  literal(security.name)
+                ));
+
+                this.quads.push(quad(
+                  securitySubject,
+                  namedNode(namespaces.idsa + 'tokenValue'),
+                  literal(security.value)
+                ));
+              } else if (security.type == 'none') {
+                this.quads.push(quad(
+                  securitySubject,
+                  namedNode(namespaces.rdf + 'type'),
+                  namedNode(namespaces.td + 'NoSecurityScheme')
+                ));
+              }
+            });
+          } else if (source.security === 'none') {
```

Comment:
This should be tackled in the expander, not here.

Reply:
Fixed! I also added a new test case with the iRail API and tested the generated RML file with the RMLMapper.
Found a bug in the meantime so that's fixed as well :)